bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,494 @@
+                        case "hours":
+                            msg.setFeatureName(featureName + " Hours");
+                            msg.setFeature(featureDataDTO.feature + "#hours");
+                            typeEntry = prop.hours.type;
+                            valueEntry = prop.hours.value.toString();
+                            break;
+                        default:
+                            break;
+                    }
+                    msg.setType(typeEntry);
+                    msg.setValue(valueEntry);
+                    msg.setChannelType("type-" + typeEntry);
+                    Boolean active = true;
+                    if (msg.getDeviceId().indexOf(config.deviceId) != -1 && active) {
+                        logger.trace("Feature: {} Type:{} Entry: {}={}", featureDataDTO.feature, typeEntry, entry,
+                                valueEntry);
+                        if (thing.getChannel(msg.getChannelId()) == null && !"unit".equals(entry)) {
+                            createChannel(msg);
+                        }
+                        if ("temperature".equals(typeEntry)) {
+                            DecimalType state = DecimalType.valueOf(msg.getValue());
+                            updateState(msg.getChannelId(), state);
+                        } else if ("number".equals(typeEntry)) {
+                            DecimalType state = DecimalType.valueOf(msg.getValue());
+                            updateState(msg.getChannelId(), state);
+                        } else if ("boolean".equals(typeEntry)) {
+                            OnOffType state = bool ? OnOffType.ON : OnOffType.OFF;
+                            updateState(msg.getChannelId(), state);
+                        } else if ("string".equals(typeEntry) || "Schedule".equals(typeEntry)
+                                || "array".equals(typeEntry)) {
+                            StringType state = StringType.valueOf(msg.getValue());
+                            updateState(msg.getChannelId(), state);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates new channels for the thing.
+     *
+     * @param msg contains everything is needed of the channel to be created.
+     */
+    private void createChannel(ThingMessageDTO msg) {
+        ChannelUID channelUID = new ChannelUID(thing.getUID(), msg.getChannelId());
+        ThingHandlerCallback callback = getCallback();
+        if (callback == null) {
+            logger.warn("Thing '{}'not initialized, could not get callback.", thing.getUID());
+            return;
+        }
+
+        Map<String, String> prop = new HashMap<>();
+        prop.put("feature", msg.getFeatureClear());
+        String channelType = msg.getChannelType();
+
+        FeatureCommands commands = msg.getCommands();
+        if (commands != null) {
+            ArrayList<String> com = commands.getUsedCommands();
```

Comment:
```suggestion
            List<String> com = commands.getUsedCommands();
```

Use the interface if a special implementation is not needed.